src/_pytest/python.py

```diff
@@ -1453,6 +1453,12 @@ def __init__(
         if callobj is not NOTSET:
             self.obj = callobj
 
+        #: Original function name, without any decorations (for example
+        #: parametrization adds a ``"[...]"`` suffix to function names).
+        #:
+        #: .. versionadded:: 3.0
+        self.originalname = originalname if originalname is not None else name
```

Comment:
Given a field called "originalname", I'd expect it to only contain the original name, but this adds a fallback to `name`, which is presumably not the original name? If so, then this seems unadvisable. Maybe do this `if` in `_getobj` instead?

Reply:
Ideally this would be a alias for functiondefinition.name

Reply:
@bluetech the point is that it should be usable as-is, i.e. also for non-parametrized functions (where `name` _is_ the original name AFAICT).
Check https://github.com/pytest-dev/pytest/commit/1a79137d04560932ac6aceccd5b20cddf106640f: it only passes it in for parametrized functions.

Reply:
(Replying only to the diff, not Ronny's comment which I didn't look into)

>  Check 1a79137: it only passes it in for parametrized functions.

Would it be possible to fix *that*? I.e. make `originalname` be passed always, even when it's the same as `name`? That would address my concern and makes some sense anyway I think...

Reply:
> Would it be possible to fix that?

Done it. 👍 

I tried @RonnyPfannschmidt's suggestion, which I think is more elegant, but the problem is that `_getobj` uses `originalname` to obtain the function, and `originalname` requires the function, so we get a recursion.